src/Storages/Distributed/DistributedSettings.h

```diff
@@ -17,6 +17,8 @@ class ASTStorage;
 #define LIST_OF_DISTRIBUTED_SETTINGS(M, ALIAS) \
     M(Bool, fsync_after_insert, false, "Do fsync for every inserted. Will decreases performance of inserts (only for background INSERT, i.e. distributed_foreground_insert=false)", 0) \
     M(Bool, fsync_directories, false, "Do fsync for temporary directory (that is used for background INSERT only) after all part operations (writes, renames, etc.).", 0) \
+    /** This is the distributed version of the skip_unavailable_shards setting available in src/Core/Settings.h */ \
+    M(Bool, skip_unavailable_shards, false, "If true, ClickHouse silently skips unavailable shards and nodes unresolvable through DNS. Shard is marked as unavailable when none of the replicas can be reached.", 0) \
```

Comment:
The comment is ambiguous. It could be read as only DNS failures are skipped. But connection failures and even shards with non-existent tables are skipped as well.

Reply:
If the original description has the same problem, it should be fixed as well.

Reply:
I have updated the comment. Please take a look and let me know if the new description is OK, or if you need this further updated. I have also updated the comment in the original description in src/Core/Settings.h.